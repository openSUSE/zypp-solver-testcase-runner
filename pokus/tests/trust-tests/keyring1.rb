#!/usr/bin/ruby


# Check if importing of a public key works.


require 'zypp'
include Zypp


keyring = ZYppFactory::instance.get_zypp.key_ring

path = Pathname.new("../../data/pubkey.txt")
publickey = PublicKey.new(path)
puts publickey.to_s()

id = publickey.id()

puts "is_key_known/trusted #{keyring.is_key_known(id)} #{keyring.is_key_trusted(id)}"

keyring.import_key(publickey, true)

puts "is_key_known/trusted #{keyring.is_key_known(id)} #{keyring.is_key_trusted(id)}"

