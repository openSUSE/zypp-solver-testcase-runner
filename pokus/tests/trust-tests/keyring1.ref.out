[1D5F7D707936F65B] [ZYpp Testsuite <none@opensuse.org>] [8F041E3F4CE465AC0D5A37001D5F7D707936F65B]
is_key_known/trusted false false
is_key_known/trusted true false
is_key_known/trusted false false
is_key_known/trusted true true
is_key_known/trusted false false
